clean-abap/CleanABAP_es.md

```diff
@@ -0,0 +1,5127 @@
+  IF sy-subrc IS INITIAL.
+    IF input = <constant>.
+      DATA(is_valid) = abap_true.
+      RETURN.
+    ENDIF.
+  ELSE.
+    RETURN.
+  ENDIF.
+ENDDO.
+```
+
+> Lee más en _Capítulo 17: Síntomas y heurística: G27: Estructura sobre convención_ de [_Código Limpio_ por Robert C. Martin].
+
+## Variables
+
+> [Clean ABAP](#clean-abap) > [Contenido](#contenido) > [Esta sección](#variables)
+
+### Prefiere declaraciones in-line que al inicio
+
+> [Clean ABAP](#clean-abap) > [Contenido](#contenido) > [Variables](#variables) > [Esta sección](#prefiere-declaraciones-in-line-que-al-inicio)
+
+Si sigues esta guía, tus métodos se volverán tan cortos (3-5 sentencias)
+que declarar variables in-line será más natural
+
+```ABAP
+METHOD do_something.
+  DATA(name) = 'something'.
+  DATA(reader) = /clean/reader=>get_instance_for( name ).
+  result = reader->read_it( ).
+ENDMETHOD.
+```
+
+que declarar variables con una sección separada de `DATA` al inicio del método
+
+```ABAP
+" anti-pattern
+METHOD do_something.
+  DATA:
+    name   TYPE seoclsname,
+    reader TYPE REF TO /dirty/reader.
+  name = 'something'.
+  reader = /dirty/reader=>get_instance_for( name ).
+  result = reader->read_it( ).
+ENDMETHOD.
+```
+
+> Lee más en _Capítulo 5: Formato vertical: Distancia Vertical: Declaraciones de variables_ de [_Código Limpio_ por Robert C. Martin].
+
+### No declares variables en ramas opcionales
+
+> [Clean ABAP](#clean-abap) > [Contenido](#contenido) > [Variables](#variables) > [Esta sección](#no-declares-variables-en-ramas-opcionales)
+
+```ABAP
+" anti-pattern
+IF has_entries = abap_true.
+  DATA(value) = 1.
+ELSE.
+  value = 2.
+ENDIF.
+```
+Esto funciona bien porque ABAP maneja declaraciones in-line como si estuvieran al inicio del método.
+Sin embargo, es extremadamente confuso para los lectores, especialmente si el método es largo y no
+detectas la declaración inmediatamente.
+En este caso, no uses in-line y coloca la declaración al inicio.
+
+```ABAP
+DATA value TYPE i.
+IF has_entries = abap_true.
+  value = 1.
+ELSE.
+  value = 2.
+ENDIF.
+```
+
+> Lee más en _Capítulo 5: Formato vertical: Distancia Vertical: Declaraciones de variables_ de [_Código Limpio_ por Robert C. Martin].
+
+### No encadenes declaraciones
+
+> [Clean ABAP](#clean-abap) > [Contenido](#contenido) > [Variables](#variables) > [Esta sección](#no-encadenes-declaraciones)
+
+```ABAP
+DATA name TYPE seoclsname.
+DATA reader TYPE REF TO reader.
+```
+El encadenamiento sugiere que las variables definidas están relacionadas en un nivel lógico.
+Para usarlo consistentemente, tienes que asegurarte de que todas las variables encadenadas
+pertenecen juntas lógicamente e introducir grupos encadenados adicionales para agregar
+variables.
+Aunque esto es posible, el esfuerzo rara vez vale la pena.
+
+El encadenamiento complica innecesariamente reformatear y refactorizar el código,
+ya que cada línea se ve diferente y cambiarlas require lidiar con 
```

Comment:
```suggestion
ya que cada línea es diferente y cambiarlas require lidiar con 
```

Reply:
I think "se ve diferente" is correct in this case, as all lines are obviously different.

Reply:
I agree that both are correct, I was going here for "is different" to highlight that a search/replace in the refactoring scenario would be tricky.

But of course it can also stay as is.

Reply:
Thanks for the suggestion. If you agree, then I'll leave it as is. I will just fix "requiere" since I just noticed it's missing an e.
```suggestion
ya que cada línea se ve diferente y cambiarlas requiere lidiar con 
```